### ISSUE TITLE
Can't overwrite design:type using @prop({ type: [XYZ] })

### ISSUE THREAD

**Issue by richard-jfc** [typegoose/typegoose#949]
## Versions

- System: linux
- NodeJS: 18.19.0
- Typescript: 5.3.3
- Compiler / Transpiler: ts-jest
- Typegoose(NPM): 12.5.0
- mongoose: 8.5.0
- mongodb: 0.0.0

## What is the Problem?

As per the [docs](https://typegoose.github.io/typegoose/versions/beta/docs/api/decorators/prop#type):
> Overwrite the type generated for the design:type reflection

You're supposed to be able to overwrite the `design:type` using `@props({ type: [Schema.Types.ObjectId], ref: ... })`, but this doesn't work.

## Code Example

```ts
    class X {
      @prop({ type: [Schema.Types.ObjectId], ref: () => X })
      x!: unknown;
    }

    const schema = buildSchema(X);

    // Logs "ObjectId", should log "Array"
    console.log(schema.paths.x.instance);
```

## Do you know *why* it happens?

It looks like the code relies on the `design:type` to work out whether the schema should be an array, rather than relying on the `prop type`.

**Comment by hasezoey** [comment]
> It looks like the code relies on the design:type to work out whether the schema should be an array, rather than relying on the prop type.

that is true and that needs to be manually set as the second parameter to `@prop`, see [Use Without "emitDecoratorMetadata"](https://typegoose.github.io/typegoose/docs/guides/use-without-emitDecoratorMetadata).

though i guess in the future it maybe should automatically detect if the `type` object returns a array (or map for that matter).

**Comment by hasezoey** [comment]
Implemented with 68e5a37f9c574178eb2757a3cd6d8a9baf22db5b, listing here as i forgot to include it in the commit message.

TL;DR: `type: [TYPE]` is now detected as `PropType.ARRAY`, if PropType is not manually set and is `PropType.NONE`.
